Add basic version of AdvancedFilters Page

Kicu opened the pull request:
### Details
 - adds very basic version of advanced filters page, to allow for further development of specific filter pages
 - only basic routing works, no data or saving form values works yet - these will be added in separate PRs, a few each. We believe that we will be able to use `FormProvider` to implement each of the sub-pages
 - right now if you go to `/search/filters` the RHP will look correctly however the page underneath RHP - Search - will look broken. That is to be expected, and it will be fixed in https://github.com/Expensify/App/issues/45027 - once we have new query functionality, it will be handled by defaults from parser

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/45026
PROPOSAL:


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

 - navigate to `/search/filters`
 - test that it renders correctly with some static temporary data
 - test that navigation within the advanced filters view works
 - on large screen the screen below _should_ be broken, that is expected temporarily

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/user-attachments/assets/950d3bff-379d-4a1f-ac6f-177274c71d52



</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>


https://github.com/user-attachments/assets/7ee0f8ee-dd55-4eb2-ac3c-772f26abd718


</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/user-attachments/assets/e8376451-18e5-4a41-ba43-9844e9809eb2



</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

Kicu committed:
Add basic skeleton for AdvancedFilters view

Kicu committed:
Add AdvancedSearchFilters component with sample static filters

Kicu committed:
Fix lint for AdvancedFilters

Kicu committed:
Update route for SearchAdvancedFiltersPage

Guccio163 committed:
date advanced filter rhp, empty

Guccio163 committed:
Merge branch 'kicu/45026-search-advanced-filters' of github.com:software-mansion-labs/expensify-app-fork into Guccio163/45026-search-advanced-filters

Guccio163 committed:
updating pr with resolving comments

Guccio163 committed:
undo the tsconfig.json change

Kicu committed:
Merge pull request #94 from software-mansion-labs/Guccio163/45026-search-advanced-filters

Date advanced filter rhp, empty

Kicu committed:
fix prettier on linkingConfig

Kicu committed:
cleanup basic skeleton for advanced filters

Kicu committed:
Merge branch 'main' into kicu/45026-search-advanced-filters

Kicu committed:
Remove consolelog

Kicu commented:
@luacmartins ready for review, will verify that android builds correctly and update the description later. Android builds always take ages to complete.

Kicu commented on src/languages/es.ts:
```diff
@@ -3560,6 +3560,13 @@ export default {
+        filters: {
+            date: {
+                before: 'Antes',
+                after: 'Después',
```
can you verify these translations @luacmartins

Kicu committed:
add small naming fixes

luacmartins commented:
@Guccio163 since you committed to this PR, can you please sign the CLA? https://github.com/Expensify/App/pull/45407#issuecomment-2233441325

luacmartins commented on src/pages/Search/AdvancedSearchFilters.tsx:
```diff
@@ -0,0 +1,53 @@
+    {
+        fieldName: 'type',
+        description: 'common.type' as const,
+        route: '/search/filters/type' as Route,
```
Same for the the date route
```suggestion
        route: ROUTES.SEARCH_ADVANCED_FILTERS_TYPE,
```

luacmartins commented on src/pages/Search/AdvancedSearchFilters.tsx:
```diff
@@ -0,0 +1,53 @@
+    {
+        fieldName: 'type',
+        description: 'common.type' as const,
+        route: '/search/filters/type' as Route,
```
Can we add the title here?
```suggestion
        route: '/search/filters/type' as Route,
        title: getFilterDisplayTitle(query, CONST.SEARCH.SYNTAX_ROOT_KEYS.TYPE),
```

luacmartins replied:
We'll probably need to move `advancedFilters` into the component if we do that, but I like keeping all the options in one place. We can also get rid of `fieldName` then.

Kicu replied:
Sure I can change it if that is your preference, although that would not be mine.

The way I did is that all the static values (that will not change) can be defined outside of the component, and we don't have to wrap them with `useMemo`, whereas the dynamic values can be created when needed inside the `map`. I like it that way, but I guess I can put it inside and memoize.

luacmartins commented on src/pages/Search/AdvancedSearchFilters.tsx:
```diff
@@ -0,0 +1,53 @@
+    return (
+        <View>
+            {advancedFilters.map((item) => {
+                const filterValue = getFilterDisplayTitle({}, item.fieldName);
```
See comment above about moving the title to the array declaration

luacmartins replied:
Agree that we should have a function for this

luacmartins commented on src/pages/Search/SearchAdvancedFiltersPage.tsx:
```diff
@@ -0,0 +1,29 @@
+import useThemeStyles from '@hooks/useThemeStyles';
+import AdvancedSearchFilters from './AdvancedSearchFilters';
+
+function SearchAdvancedFiltersPage() {
```
Hmm I wonder if we need to break this down into `AdvancedSearchFilters` and `SearchAdvancedFiltersPage`. Maybe we can consolidate them into a single `AdvancedSearchFiltersPage`?

luacmartins replied:
Sounds good, let's keep it separate then!

luacmartins commented on src/pages/Search/SearchFilters.tsx:
```diff
@@ -25,6 +25,7 @@ type SearchMenuFilterItem = {
     route: Route;
 };
 
+// Because we will add have AdvancedFilters, in future rename this component to `SearchTypeMenu|Tabs|Filters` to avoid confusion
```
We can already rename this to `SearchStatusBar`

Kicu replied:
Sorry but this can't be `SearchStatusBar` because this is the component that is currently (and for now still will be) displayed in LHN. So this is not the moment to rename it or it will confuse developers.

We should rename it I think when we are actually removing it from LHN - which this PR is not doing.
CC @adamgrzybowski @WojtekBoman

luacmartins commented on src/pages/Search/SearchFiltersDatePage.tsx:
```diff
@@ -0,0 +1,32 @@
+            <FullPageNotFoundView shouldShow={false}>
+                <HeaderWithBackButton title={translate('common.date')} />
+                <View style={[styles.flex1, styles.ph3]}>
+                    <Text>Advanced filters Date form</Text>
```
Can we add a comment mentioning that this is a placeholder and a link to https://github.com/Expensify/App/issues/45026 please?

luacmartins commented on src/pages/Search/SearchFiltersTypePage.tsx:
```diff
@@ -0,0 +1,32 @@
+            <FullPageNotFoundView shouldShow={false}>
+                <HeaderWithBackButton title={translate('common.type')} />
+                <View style={[styles.flex1, styles.ph3]}>
+                    <Text>Advanced filters Type form</Text>
```
Same here, let's add a comment for the placeholder

Guccio163 commented:
I have read the CLA Document and I hereby sign the CLA

Kicu commented on src/pages/Search/AdvancedSearchFilters.tsx:
```diff
@@ -0,0 +1,53 @@
+        <View>
+            {advancedFilters.map((item) => {
+                const filterValue = getFilterDisplayTitle({}, item.fieldName);
+                const onPress = singleExecution(() => Navigation.navigate(item.route));
```
will add

Kicu committed:
Add fixes after review to new search filters

Kicu committed:
Merge branch 'main' into kicu/45026-search-advanced-filters

Kicu committed:
Refactor advanced filters definition

Kicu commented:
@luacmartins I addressed most of your comments but responded in 2 cases why I prefer the current version
@rayane-djouah I addressed all your comments (I hope :D)
 - BUG1 - solved, I've made a mistake when defining URLS
 - BUG2 - this is known and appears on `main`. This specific animation is disabled forSafari temporarily until we find a better solution. https://github.com/Expensify/App/issues/44765

luacmartins reviewed (approved):
LGTM! Nice work!

luacmartins commented:
@rayane-djouah all yours!

rayane-d reviewed (approved):
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/10013421048) to staging by https://github.com/luacmartins in version: 9.0.10-2 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/10041305614) to staging by https://github.com/mountiny in version: 9.0.10-3 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/10054903341) to production by https://github.com/mountiny in version: 9.0.10-7 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/10087411925) to production by https://github.com/francoisl in version: 9.0.11-5 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
